File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/RawBatch40.java
package org.akoshterek.backgammon.agent;

import org.akoshterek.backgammon.Constants;
import org.akoshterek.backgammon.agent.fa.SimpleEncogFA;
import org.akoshterek.backgammon.agent.inputrepresentation.GnuBgCodec;
import org.akoshterek.backgammon.agent.inputrepresentation.Tesauro89Codec;
import org.akoshterek.backgammon.agent.raw.RawRepresentation;
import org.akoshterek.backgammon.board.Board;
import org.akoshterek.backgammon.eval.Reward;
import org.akoshterek.backgammon.util.Normalizer;

import java.nio.file.Path;

/**
 * @author Alex
 *         date 29.11.2015.
 */
public class RawBatch40 extends AbsFlexAgent {
    public RawBatch40(Path path) {
        super(path);
        fullName = "RawBatch40";

        contactRepresentation = new RawRepresentation(new Tesauro89Codec());
        raceRepresentation = new RawRepresentation(new GnuBgCodec());
        crashedRepresentation = new RawRepresentation(new GnuBgCodec());

        contactFa = new SimpleEncogFA(SimpleEncogFA.loadNNFromResource("org/akoshterek/backgammon/agent/raw/Raw-Tesauro89-40-contact.eg"));
        raceFa = new SimpleEncogFA(SimpleEncogFA.loadNNFromResource("org/akoshterek/backgammon/agent/raw/Raw-GnuBg-40-race.eg"));
        crashedFa = new SimpleEncogFA(SimpleEncogFA.loadNNFromResource("org/akoshterek/backgammon/agent/raw/Raw-GnuBg-40-crashed.eg"));
    }

    @Override
    public Reward evalContact(Board board) {
        Reward reward = super.evalContact(board);
        Normalizer.fromSmallerSigmoid(reward.data, Constants.NUM_OUTPUTS);
        return reward;
    }

    @Override
    public Reward evalRace(Board board) {
        Reward reward = super.evalRace(board);
        Normalizer.fromSmallerSigmoid(reward.data, Constants.NUM_OUTPUTS);
        return reward;
    }

    @Override
    public Reward evalCrashed(Board board) {
        Reward reward = super.evalCrashed(board);
        Normalizer.fromSmallerSigmoid(reward.data, Constants.NUM_OUTPUTS);
        return reward;
    }
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/raw/RawBatch40.scala
package org.akoshterek.backgammon.agent.raw

import java.nio.file.Path

import org.akoshterek.backgammon.agent.AbsFlexAgent
import org.akoshterek.backgammon.agent.fa.SimpleEncogFA
import org.akoshterek.backgammon.agent.inputrepresentation.{GnuBgCodec, Tesauro89Codec}
import org.akoshterek.backgammon.board.Board
import org.akoshterek.backgammon.eval.Reward

class RawBatch40(override val path: Path) extends AbsFlexAgent("RawBatch40", path) {
  contactRepresentation = new RawRepresentation(Tesauro89Codec)
  raceRepresentation = new RawRepresentation(GnuBgCodec)
  crashedRepresentation = new RawRepresentation(GnuBgCodec)

  contactFa = new SimpleEncogFA(SimpleEncogFA.loadNNFromResource("org/akoshterek/backgammon/agent/raw/Raw-Tesauro89-40-contact.eg"))
  raceFa = new SimpleEncogFA(SimpleEncogFA.loadNNFromResource("org/akoshterek/backgammon/agent/raw/Raw-GnuBg-40-race.eg"))
  crashedFa = new SimpleEncogFA(SimpleEncogFA.loadNNFromResource("org/akoshterek/backgammon/agent/raw/Raw-GnuBg-40-crashed.eg"))

  override def evalContact(board: Board): Reward = {
    super.evalContact(board)
  }

  override def evalRace(board: Board): Reward = {
    super.evalRace(board)
  }

  override def evalCrashed(board: Board): Reward = {
    super.evalCrashed(board)
  }
}
